src/librustc_lint/unused.rs

```diff
@@ -179,9 +180,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
                 let def_id = def.def_id();
                 fn_warned = check_must_use(cx, def_id, s.span, "return value of ");
             }
+
+            if let hir::ExprBinary(bin_op, ..) = expr.node {
+                match bin_op.node {
+                    hir::BiEq | hir::BiLt | hir::BiLe | hir::BiNe | hir::BiGe | hir::BiGt => {
```

Comment:
I think it would be less surprising, if instead of whitelisting operators, we'd check whether the corresponding operator method has the attribute.